tests/components/version/test_sensor.py

```diff
@@ -1,26 +1,56 @@
 """The test for the version sensor platform."""
 from unittest.mock import patch
 
+from pyhaversion import HaVersionSource, exceptions as pyhaversionexceptions
+import pytest
+
+from homeassistant.components.version.sensor import ALL_SOURCES
 from homeassistant.setup import async_setup_component
 
 MOCK_VERSION = "10.0"
 
 
-async def test_version_sensor(hass):
-    """Test the Version sensor."""
-    config = {"sensor": {"platform": "version"}}
+@pytest.mark.parametrize(
+    "source",
+    ALL_SOURCES,
+)
+async def test_version_source(hass, source):
+    """Test the Version sensor with different sources."""
+    config = {
+        "sensor": {"platform": "version", "source": source, "image": "qemux86-64"}
+    }
+
+    with patch("pyhaversion.version.HaVersion.version", MOCK_VERSION):
+        assert await async_setup_component(hass, "sensor", config)
+        await hass.async_block_till_done()
 
-    assert await async_setup_component(hass, "sensor", config)
+    name = "current_version" if source == HaVersionSource.LOCAL else "latest_version"
```

Comment:
I'd parametrize name too besides source to avoid the logic check here.

Reply:
Good idea :+1:

Reply:
Actually, this was done to test the handling of the default name.
https://github.com/home-assistant/core/blob/39d7bb4f1a3711fb9ccc5c3f74efa021e0c10adc/homeassistant/components/version/sensor.py#L96-L100

Reply:
Yeah, it would be annoying to manually define the sources since some are imported from the library. Then it's just a slight improvement to move the if logic to a list comprehension in the parametrization decorator. So maybe not worth it.

Reply:
It was doing I/O so had to change it anyway <https://github.com/home-assistant/core/pull/54391>